packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,20 +142,95 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      let server: http.Server;
+
+      if (httpsSettings) {
+        logger.info('Initializing https server');
+
+        const credentials: { key: string; cert: string } = {
+          key: '',
+          cert: '',
+        };
+        const signingOptions: any = httpsSettings?.certificate;
+
+        if (signingOptions?.algorithm !== undefined) {
+          logger.info('Generating self-signed certificate with attributes');
+
+          const certificateAttributes: Array<any> = [];
+
+          Object.entries(signingOptions?.attributes).map(([name, value]) => {
```

Comment:
Better use `forEach` or a `for (... of ...)` loop when you're not using the return value of the iterator function.

Reply:
You asked me to change it to this variant two days ago (check resolved comments). Do you want me to replace the .map with a forEach? Its not a problem, but I figured I would just use your suggestions AS-IS? :)

Reply:
`Object.entries(signingOptions.attributes).map(([name, value]) => ({ name, value }))` is what I suggested, plain functional style. Imperative style works too but it's a bit cleaner to not use `map` in that case.

Reply:
With my limited understanding of NodeJS/TS it seems to me like the original suggestion did not really take into account that we needed to push the mapped value to the certAttributes array. Which is why I augmented the map function to handle this. Now if where is a cleaner way to do it, then what I just implemented, I have no problem refactoring it to meet that requirement. However as it stands I cannot revert to your AS-IS suggestion without modification because in my understanding that stop the propagation of values to the certAttr array and break the self-signed signature generation.